src/conn/conn_log.c

```diff
@@ -793,8 +817,8 @@ __log_server(void *arg)
 		}
 
 		/* Wait until the next event. */
-		WT_ERR(__wt_cond_wait_signal(session, conn->log_cond,
-		    WT_MILLION / WT_FORCE_PER_SECOND, &signalled));
+		WT_ERR(__wt_cond_auto_wait_signal(session, conn->log_cond,
+		    did_work || signalled, &signalled));
```

Comment:
I don't think you need signalled in the condition - the auto conditionals clear the timer automatically when woken by a signal.